src/components/Modal.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react";
+import { ImCross } from "react-icons/im";
+import { GrEmoji, GrImage } from "react-icons/gr";
+import { useDispatch, useSelector } from "react-redux";
+import { closeModal } from "../features/modalSlice";
+import { addPost, editPost } from "../features/postSlice";
+
+export const Modal = () => {
+  const dispatch = useDispatch();
+  const { token } = useSelector((state) => state.auth);
+  const { setContent } = useSelector((state) => state.modal);
+  const [postInput, setPostInput] = useState("");
+  const { content, _id } = setContent;
+
+  useEffect(() => {
+    setPostInput(content);
+  }, [content]);
+
+  const modalCloseHandler = () => {
+    dispatch(closeModal());
+  };
+
+  const postHandler = (e) => {
+    setPostInput(e.target.value);
+  };
+
+  const addPostHandler = () => {
+    dispatch(addPost({ postInput, token }));
+    dispatch(closeModal());
+  };
+
+  const updatePostHandler = () => {
+    dispatch(editPost({ postInput, token, _id }));
+    dispatch(closeModal());
+  };
+
+  const cancelModal = () => {
+    dispatch(closeModal());
+  };
```

Comment:
Can remove this function as you have defined a function that performs the same action named `modalCloseHandler`.

Reply:
i may need to perform some different thing later, if that will not be the case I will remove later